src/components/molecules/ChatList/ChatList.tsx

```diff
@@ -84,20 +87,26 @@ const ChatList: React.FC<ChatListProps> = ({
   const hideDeleteModal = useCallback(() => setShowDeleteModal(false), []);
 
   const hasMessages = hasElements(messages);
+
   const messageSender = usersById?.[messageToDelete?.from ?? ""]?.partyName;
+
+  // Last (newest) message goes first
+  const sortedMessages = sortBy(messages, ["ts_utc"]).reverse();
+
   return (
     <>
       {hasMessages && (
         <div className="chat-messages-container">
           {usersById &&
-            messages.map((message, index) => (
+            sortedMessages.map((message, index) => (
               <ChatMessage
                 key={`chat-message-${index}`}
                 usersById={usersById}
                 message={message}
                 allowDelete={allowDelete ?? false}
                 onDeleteClick={toggleDeleteModal}
                 onAvatarClick={showUserProfile}
+                showSenderImage={showSenderImage}
               />
             ))}
         </div>
```

Comment:
Extract this from the JSX + `useMemo`. If it's the only place `sortedMessages` is needed/used, then you could probably combine this + my above comment within the same `useMemo` block.

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697